jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -222,15 +239,21 @@ public static String getErrorMessage() {
 	/* Higher priority hooks are run last in pre-checkoint hooks, and are run
 	 * first in post restore hooks.
 	 */
-	private static final int RESTORE_CLEAR_INETADDRESS_CACHE_PRIORITY = 100;
-	private static final int RESTORE_ENVIRONMENT_VARIABLES_PRIORITY = 100;
-	private static final int RESTORE_SYSTEM_PROPERTIES_PRIORITY = 100;
-	private static final int USER_HOOKS_PRIORITY = 1;
+	// the lowest priority of a CONCURRENT_MODE hook
+	public static int CONCURRENT_MODE_HOOK_PRIORITY_LOWEST = 0;
```

Comment:
these can just be renamed to `USER_HOOK_PRIORITY_LOWEST`

Reply:
How about following priority constants?
```
	// the lowest priority of a user hook
	public static int LOWEST_USER_HOOK_PRIORITY = 0;
	// the highest priority of a user hook
	public static int HIGHEST_USER_HOOK_PRIORITY = 99;
	// the default SINGLE_THREAD_MODE hook priority
	private static final int DEFAULT_SINGLE_THREAD_MODE_HOOK_PRIORITY = LOWEST_USER_HOOK_PRIORITY + 1;
	// other SINGLE_THREAD_MODE hook priority
	private static final int RESTORE_CLEAR_INETADDRESS_CACHE_PRIORITY = HIGHEST_USER_HOOK_PRIORITY + 1;
	private static final int RESTORE_ENVIRONMENT_VARIABLES_PRIORITY = HIGHEST_USER_HOOK_PRIORITY + 1;
	private static final int RESTORE_SYSTEM_PROPERTIES_PRIORITY = HIGHEST_USER_HOOK_PRIORITY + 1;
	/* RESET_CRIUSEC_PRIORITY and RESTORE_SECURITY_PROVIDERS_PRIORITY need to
	 * be higher than any other JVM hook that may require security providers.
	 */
	static final int RESET_CRIUSEC_PRIORITY = HIGHEST_USER_HOOK_PRIORITY + 1;
	static final int RESTORE_SECURITY_PROVIDERS_PRIORITY = HIGHEST_USER_HOOK_PRIORITY + 1;
```
Application invoking CRIU APIs specifies a priority between `LOWEST_USER_HOOK_PRIORITY` and `HIGHEST_USER_HOOK_PRIORITY` for either `SINGLE_THREAD_MODE` or `CONCURRENT_MODE`.
The existing `registerPreCheckpointHook(Runnable)` and `registerPostRestoreHook(Runnable)` convenient APIs specify `DEFAULT_SINGLE_THREAD_MODE_HOOK_PRIORITY` (`SINGLE_THREAD_MODE`).
The hooks registered by JVM itself (only `SINGLE_THREAD_MODE` so far) are `HIGHEST_USER_HOOK_PRIORITY + 1`.